presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java

```diff
@@ -248,57 +276,104 @@ private static ConnectorTableMetadata getTableMetadata(String schemaName, TpchTa
-                        .orElse(ORDER_STATUS_VALUES);
-
-                long totalRows = 0L;
-                if (orderStatusValues.contains("F")) {
-                    totalRows = 729_413;
-                }
-                if (orderStatusValues.contains("O")) {
-                    totalRows += 732_044;
-                }
-                if (orderStatusValues.contains("P")) {
-                    totalRows += 38_543;
-                }
-                return (long) (totalRows * scaleFactor);
-            case "lineitem":
-                return (long) (6_000_000 * scaleFactor);
-            case "part":
-                return (long) (200_000 * scaleFactor);
-            case "partsupp":
-                return (long) (800_000 * scaleFactor);
-            case "supplier":
-                return (long) (10_000 * scaleFactor);
-            case "nation":
-                return 25;
-            case "region":
-                return 5;
-            default:
-                throw new IllegalArgumentException("unknown tpch table name '" + tableName + "'");
+        TupleDomain<ColumnHandle> constraintSummary = constraint.getSummary();
+        if (constraintSummary.isAll()) {
+            return emptyMap();
         }
+        else if (constraintSummary.isNone()) {
+            Set<TpchColumn<?>> columns = ImmutableSet.copyOf(tpchTable.getColumns());
+            return asMap(columns, key -> emptyList());
+        }
+        else {
+            Map<ColumnHandle, Domain> domains = constraintSummary.getDomains().get();
+            Optional<Domain> orderStatusDomain = Optional.ofNullable(domains.get(toColumnHandle(ORDER_STATUS)));
+            Optional<Map<TpchColumn<?>, List<Object>>> allowedColumnValues = orderStatusDomain.map(domain -> {
+                List<Object> allowedValues = ORDER_STATUS_VALUES_SLICES.stream()
+                        .filter(domain::includesNullableValue)
+                        .collect(toList());
+                return avoidTrivialOrderStatusRestriction(allowedValues);
+            });
+            return allowedColumnValues.orElse(emptyMap());
+        }
+    }
+
+    private Map<TpchColumn<?>, List<Object>> avoidTrivialOrderStatusRestriction(List<Object> allowedValues)
+    {
+        if (allowedValues.containsAll(ORDER_STATUS_VALUES_SLICES)) {
+            return emptyMap();
+        }
+        else {
+            return ImmutableMap.of(ORDER_STATUS, allowedValues);
+        }
+    }
+
+    private TableStatistics toTableStatistics(TableStatisticsData tableStatisticsData, TpchTableHandle tpchTableHandle, Map<String, ColumnHandle> columnHandles)
+    {
+        TableStatistics.Builder builder = TableStatistics.builder()
+                .setRowCount(new Estimate(tableStatisticsData.getRowCount()));
+        tableStatisticsData.getColumns().forEach((columnName, stats) -> {
+            TpchColumnHandle columnHandle = (TpchColumnHandle) getColumnHandle(tpchTableHandle, columnHandles, columnName);
```

Comment:
What about different column naming schemas in TPCH connector. Does it work with both?